speakers/src/main/java/in/droidcon/speakers/domain/SpeakerRepositoryImplementation.kt

```diff
@@ -1,49 +1,40 @@
 package `in`.droidcon.speakers.domain
 
-import `in`.droidcon.base.core.getOneSpeaker
-import `in`.droidcon.base.core.getSpeakerList
-import `in`.droidcon.data.speakers.model.SpeakerEntity
-import com.google.firebase.firestore.FirebaseFirestore
+import `in`.droidcon.speakers.model.SpeakerEntity
+import `in`.droidcon.speakers.remote.SpeakerMapper
+import `in`.droidcon.speakers.remote.SpeakerService
 import io.reactivex.Single
+import org.koin.standalone.KoinComponent
+import org.koin.standalone.inject
 
 /**
  * @author Adnan A M
  * @since  20/03/19
  */
-class SpeakerRepositoryImplementation: SpeakerRepository {
+class SpeakerRepositoryImplementation : SpeakerRepository, KoinComponent {
 
-    private val fireStore: FirebaseFirestore = FirebaseFirestore.getInstance()
+    private val speakerService: SpeakerService by inject()
+
+    val speakersCache: List<SpeakerEntity>? = null
 
     override fun getSpeakers(): Single<List<SpeakerEntity>> {
-        return Single.create<List<SpeakerEntity>> { emitter ->
-            fireStore.getSpeakerList()
-                .addOnCompleteListener { task ->
-                    if (task.isSuccessful) {
-                        task.result?.let { result ->
-                            val list: List<SpeakerEntity> = result.toObjects(SpeakerEntity::class.java)
-                            emitter.onSuccess(list)
-                        } ?: emitter.onSuccess(listOf())
-                    } else {
-                        emitter.onError(Throwable(task.exception))
-                    }
-                }
+        return if (speakersCache == null) {
```

Comment:
According to current implementation, `speakerCache` will be always `null` are we planning to populate it from persistence? In that case, should we consider persistence as the single source of truth? What about update policy? will it be a push or pull mechanism? Thoughts?

Reply:
@iamBedant Thanks for this, I totally missed saving the `speakersList` to `speakersCache`.
The idea here is to save the speakers list in memory, since these are not expected to change very often, at least in a single lifetime of the fragment.
However, our primary goal is to save the data in Room/SQLite/SQLDelight after fetching it for the first time, I couldn't figure out the proper way to configure Room yet for clean architecture where multiple modules have dependencies (SQL) on each other.
Please do it if you can.

Reply:
Saved speakerList in speakerCache